### ISSUE TITLE
Command "setup:db-declaration:generate-patch" does not add revert() method to patch

### ISSUE THREAD

**Issue by atwixfirster** [magento/magento2#23847]
### Summary (*)

Command [setup:db-declaration:generate-patch](https://devdocs.magento.com/guides/v2.3/reference/cli/magento.html#setupdb-declarationgenerate-patch) generates a patch file without `revert()` method even when `--revertable` option is set to `true`.

### Examples (*)

Run 
```
bin/magento setup:db-declaration:generate-patch --revertable=true Vendor_Module TestPatch
```

`Vendor_Module` - any module. 

Example with `Magento_Cms`:
```
bin/magento setup:db-declaration:generate-patch --revertable=true Magento_Cms TestPatch
```

**Actual result:** 
`app/code/Magento/Cms/Setup/Patch/Data/TestPatch.php` is created with content without `revert()` method:

```php
<?php
/**
 * Copyright © Magento, Inc. All rights reserved.
 * See COPYING.txt for license details.
 */

namespace Magento\Cms\Setup\Patch\Data;

use Magento\Framework\Setup\Patch\DataPatchInterface;
use Magento\Framework\Setup\Patch\SchemaPatchInterface;
use Magento\Framework\Setup\Patch\PatchRevertableInterface;
use Magento\Framework\Setup\ModuleDataSetupInterface;

/**
* Patch is mechanism, that allows to do atomic upgrade data changes
*/
class TestPatch implements
    DataPatchInterface
{
    /**
     * @var ModuleDataSetupInterface $moduleDataSetup
     */
    private $moduleDataSetup;

    /**
     * @param ModuleDataSetupInterface $moduleDataSetup
     */
    public function __construct(ModuleDataSetupInterface $moduleDataSetup)
    {
        $this->moduleDataSetup = $moduleDataSetup;
    }

    /**
     * Do Upgrade
     *
     * @return void
     */
    public function apply()
    {
    }

    /**
     * {@inheritdoc}
     */
    public function getAliases()
    {
        return [];
    }

    /**
     * {@inheritdoc}
     */
    public static function getDependencies()
    {
        return [

        ];
    }
}
```

**Expected result:**
`app/code/Magento/Cms/Setup/Patch/Data/TestPatch.php` is created with `revert()` method in the content:

```php
<?php
/**
 * Copyright © Magento, Inc. All rights reserved.
 * See COPYING.txt for license details.
 */

namespace Atwix\ShippingTableRates\Setup\Patch\Schema;

use Magento\Framework\Setup\Patch\DataPatchInterface;
use Magento\Framework\Setup\Patch\SchemaPatchInterface;
use Magento\Framework\Setup\Patch\PatchRevertableInterface;
use Magento\Framework\Setup\ModuleDataSetupInterface;

/**
* Patch is mechanism, that allows to do atomic upgrade data changes
*/
class Add2 implements
    SchemaPatchInterface, PatchRevertableInterface
{
    /**
     * @var ModuleDataSetupInterface $moduleDataSetup
     */
    private $moduleDataSetup;

    /**
     * @param ModuleDataSetupInterface $moduleDataSetup
     */
    public function __construct(ModuleDataSetupInterface $moduleDataSetup)
    {
        $this->moduleDataSetup = $moduleDataSetup;
    }

    /**
     * Do Upgrade
     *
     * @return void
     */
    public function apply()
    {
    }

    /**
     * @inheritdoc
     */
    public function revert()
    {
    }

    /**
     * {@inheritdoc}
     */
    public function getAliases()
    {
        return [];
    }

    /**
     * {@inheritdoc}
     */
    public static function getDependencies()
    {
        return [

        ];
    }
}
```

### Proposed solution

Fix provided in the PR: https://github.com/magento/magento2/pull/23848

**Comment by magento-engcom-team** [comment]
:white_check_mark: Confirmed by @dmytro-ch 
 Thank you for verifying the issue. Based on the provided information internal tickets `MC-18769` were created

**Issue Available**: @dmytro-ch, _You will be automatically unassigned. Contributors/Maintainers can claim this issue to continue. To reclaim and continue work, reassign the ticket to yourself._

**Comment by magento-engcom-team** [comment]
Hi @atwixfirster. Thank you for your report. 
The issue has been fixed in magento/magento2#23848 by @atwixfirster in 2.3-develop branch
Related commit(s):
 - [84527e712a49bced109e784978dcb3f8be391246](https://github.com/magento/magento2/commit/84527e712a49bced109e784978dcb3f8be391246)

The fix will be available with the upcoming 2.3.4 release.
